samples/python/refresh-authors.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/python
+
+###############################################################################
+# ex. python refresh-authors.py --conf MyConf.org/2017 --baseurl http://localhost:3000
+#       --username admin --password admin_pw
+#
+# To be run to refresh author groups in case of revisions and/or withdrawals
+# Expects config to define BLIND_SUBMISSION, SUBMISSION and AUTHORS as well
+# as an Authors group for each SUBMISSION paper
+###############################################################################
+
+## Import statements
+import argparse
+import sys
+from openreview import *
+
+## Argument handling
+parser = argparse.ArgumentParser()
+parser.add_argument('-c','--conf', required=True, help = "the full path of the conference group ex. MyConf.org/2017")
+parser.add_argument('--baseurl', help="base url")
+parser.add_argument('--username')
+parser.add_argument('--password')
+
+args = parser.parse_args()
+
+## Initialize the client library with username and password
+client = Client(baseurl=args.baseurl, username=args.username, password=args.password)
+
+## check conference directory exists
+base_path = "../../venues/"+args.conf
+config_path = base_path+"/python/"
+if os.path.isfile(config_path+"config.py") is False:
+    print "Cannot locate config.py in:"+config_path
+    sys.exit()
+## load conference specific data
+sys.path.insert(0, config_path)
+import config
+
+### update authors lists
+all_authors = []
+submissions = client.get_notes(invitation=config.SUBMISSION)
+blinded = client.get_notes(invitation=config.BLIND_SUBMISSION)
+for blind_note in blinded:
+    #find associated note in Submissions to get true authors
+    parent_forum = blind_note.original
+    note_authors = []
+    found_profile = False
+    #for parent_notes = submissions[].forum == note.original
+    for note in submissions:
```

Comment:
instead of looping through all submissions and breaking when you find the right one, you could instead create a dictionary of forum IDs to submissions, and then index into it using the blind_note.original value.

so line 41 could be something like:

`submissions_by_forum = {n.forum: n for n in client.get_notes(invitation=config.SUBMISSION)`

Then, in this loop, you could look up the note by doing:

`note = submissions_by_forum[blind_note.original]`

Reply:
Yes, using dictionaries is cheaper than loop the all the notes every time.